.github/workflows/tf_apply_production.yml

```diff
@@ -0,0 +1,45 @@
+name: "Terraform Apply Production"
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "terragrunt/**"
+      - "!terragrunt/env/production/**"
```

Comment:
You can remove this path since we'll also want to run this workflow if there are changes in this folder.